### ISSUE TITLE
Navtitle specified in topicref metadata seems to override title specified in topic [DOT 2.x]

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2187]
I have a topicref like:

``` xml
<topicref href="my_service.dita">
  <topicmeta>
    <navtitle>SHOULD NOT OVERRIDE</navtitle>
  </topicmeta>
</topicref>
```

and the published XHTML output should ignore the navtitle (because the @locktitle attribute has not been set) and use instead the title from the topic.
But somehow with DITA OT 2.x the title displayed in the table of contents is taken from the topicmeta.
This works properly with DITA OT 1.8.

**Comment by jelovirt** [comment]
Is the issue that `navtitle` from map is used instead of `navtitle` in topic?

**Comment by raducoravu** [comment]
The issue is that the navtitle element from the topicmeta is used instead of the topic title element. Even though the @locktitle is not set (and thus the title is not locked).
I cannot reproduce the issue if instead of topicmeta with navtitle I'm setting the navtitle as an attribute on the topicref.

**Comment by eerohele** [comment]
Introduced in 77d3cd64eccf66868753f4853c7f86117daaa8bd.

**Comment by raducoravu** [comment]
We had one more user reporting this problem. I added an extra vote on the issue.

**Comment by raducoravu** [comment]
Another one of our users encountered this problem. As an extra behavior, when the topic also specifies a navtitle in its titlealts element:

    <titlealts>
        <navtitle>Topic Overview</navtitle>
        <searchtitle>Overview</searchtitle>
    </titlealts>

the table of contents for the XHTML output will concatenate the navtitle specified in the DITA Map with the navtitle set inside the topic.

**Comment by robander** [comment]
Fixed with #2897
